src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-chart/cephfs-chart.component.ts

```diff
@@ -22,136 +23,140 @@ export class CephfsChartComponent implements OnChanges, OnInit {
   lhsCounter = 'mds_mem.ino';
   rhsCounter = 'mds_server.handle_client_request';
 
-  chart: any;
+  chart = {
+    datasets: [
+      {
+        label: this.lhsCounter,
+        yAxisID: 'LHS',
+        data: [] as ChartPoint[],
+        lineTension: 0.1
+      },
+      {
+        label: this.rhsCounter,
+        yAxisID: 'RHS',
+        data: [] as ChartPoint[],
+        lineTension: 0.1
+      }
+    ] as ChartDataSets[],
+    options: {
+      title: {
+        text: '',
+        display: true
+      },
+      responsive: true,
+      maintainAspectRatio: false,
+      legend: {
+        position: 'top'
+      },
+      scales: {
+        xAxes: [
+          {
+            position: 'top',
+            type: 'time',
+            time: {
+              displayFormats: {
+                quarter: 'MMM YYYY'
+              }
+            },
+            ticks: {
+              maxRotation: 0
+            }
+          }
+        ],
+        yAxes: [
+          {
+            id: 'LHS',
+            type: 'linear',
+            position: 'left'
+          },
+          {
+            id: 'RHS',
+            type: 'linear',
+            position: 'right'
+          }
+        ]
+      },
+      tooltips: {
+        enabled: false,
+        mode: 'index',
+        intersect: false,
+        position: 'nearest',
+        callbacks: {
+          // Pick the Unix timestamp of the first tooltip item.
+          title: function(tooltipItems, data): string {
+            let ts = 0;
+            if (tooltipItems.length > 0) {
+              const item = tooltipItems[0];
+              const point = data.datasets[item.datasetIndex].data[item.index] as ChartPoint;
+              ts = point.x as number;
```

Comment:
If `point` is of type `ChartPoint`, does `point.x` really need the `as number` addition?

Reply:
As `x` of `CharPoint` can be `number`, `string` or `Date`, `as number` is needed in order to assign it to the variable ts which holds a number.

Before the use of types here the code looked like this:
```ts
              let ts = 0;
              if (tooltipItems.length > 0) {
                const item = tooltipItems[0];
                ts = data.datasets[item.datasetIndex].data[item.index].x;
              }
              return ts;
```
But that will produce compile errors if types are in use.